### PR TITLE
LLM: Enable Speculative on Fastchat

### PR DESCRIPTION
**Pull request by hzjane**
## Description
Enable Speculative on Fastchat
Llama-7b speculative will use near 16.5GB（fp16+int4） memory on Self-Speculative case，So just can run on max GPU.

### 4. How to test?
- [x] Application test on spr14
- [x] Application test on max1100

### 5. New dependencies

- [ ] New Python dependencies
       - Dependency1 
       - Dependency2
       - ...
- [ ] New Java/Scala dependencies and their license
       - Dependency1 and license1
       - Dependency2 and license2
       - ...

### CODE REVIEW

File: python/llm/src/ipex_llm/serving/fastchat/README.md [hunk 1]
```diff
@@ -51,7 +53,7 @@ python3 -m fastchat.serve.controller
 
 Using IPEX-LLM in FastChat does not impose any new limitations on model usage. Therefore, all Hugging Face Transformer models can be utilized in FastChat.
 
-#### IPEX-LLM model worker (deprecated)
+#### OLD LLM model worker (deprecated)
```
Comment (glorysdj) [inline python/llm/src/ipex_llm/serving/fastchat/README.md:56]: deprecated part can be deleted

**Comment by glorysdj** [comment]
also update https://github.com/intel-analytics/ipex-llm/blob/main/docs/readthedocs/source/doc/LLM/Quickstart/fastchat_quickstart.md to add speculative support

File: docs/readthedocs/source/doc/LLM/Quickstart/fastchat_quickstart.md [hunk 2]
```diff
@@ -61,6 +61,15 @@ export SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=1
 python3 -m ipex_llm.serving.fastchat.ipex_llm_worker --model-path REPO_ID_OR_YOUR_MODEL_PATH --low-bit "sym_int4" --trust-remote-code --device "xpu"
 ```
 
+For self-speculative decoding example:
+
+You can use IPEX-LLM to run `self-speculative decoding` example. 'Refer to [here](https://github.com/intel-analytics/ipex-llm/tree/c9fac8c26bf1e1e8f7376fa9a62b32951dd9e85d/python/llm/example/GPU/Speculative-Decoding) for more details on intel MAX GPUs. Refer to [here](https://github.com/intel-analytics/ipex-llm/tree/c9fac8c26bf1e1e8f7376fa9a62b32951dd9e85d/python/llm/example/GPU/Speculative-Decoding) for more details on intel CPUs.
+
+```bash
+# Available low_bit format including bf16 on CPU, fp16 on MAX GPU.
+python3 -m ipex_llm.serving.fastchat.ipex_llm_worker --model-path lmsys/vicuna-7b-v1.5 --low-bit "bf16" --trust-remote-code --device "cpu" --speculative
```
Comment (glorysdj) [inline docs/readthedocs/source/doc/LLM/Quickstart/fastchat_quickstart.md:70]: also add gpu example

**Review by gc-fu — APPROVED** [review]
The code changes LGTM

**Pull request merged**